.github/workflows/linter.yml

```diff
@@ -0,0 +1,47 @@
+name: Linter check
+
+env:
+  HUSKY: 0
+
+on:
+  pull_request:
+    branches: ["**"]
+  workflow_dispatch:
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js 22
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: |
+          corepack enable
+          yarn install
+
+      - name: Check there are no errors to be fixed in package.json and no uncommitted changes
+        run: |
+          npm pkg fix && git diff --exit-code
+
+      - name: Check there are no errors reported by ESLint
+        run: |
+          yarn lint
+
+      - name: Check formatting with Prettier
+        run: |
+          yarn fmt:check
+
+      - name: Spellcheck code base
+        run: |
+          yarn spell
+
+      - name: Check there are no unused dependencies
+        run: |
+          yarn knip
```

Comment:
Let's run all these steps (even if some previous fail) and report CI failure if any of those fails.

Reply:
I think how to do it better, there are two realizations:
1) Through `matrix`, but then it will be necessary to put dependencies on each task. This is not optimized;
2) Add for each step `continue-on-error: true`, then the failed steps will not be shown, it can be considered that it is equal to `|| true`.